src/expr/src/scalar/func.rs

```diff
@@ -991,6 +991,24 @@ fn sqrt_float64<'a>(a: Datum<'a>) -> Datum<'a> {
     Datum::from(x.sqrt())
 }
 
+fn sqrt_dec<'a>(a: Datum<'a>, scale: u8) -> Datum<'a> {
```

Comment:
Can you make this return a `Result<Datum, EvalError>` and then just propagate the error from `cast_float64_to_decimal`?